controllers/validator.go

```diff
@@ -65,26 +75,50 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
 		return false, err
 	}
 
-	if _, err := getResticResourceReqs(dpa); err != nil {
+	if _, err := getResticResourceReqs(r.dpa); err != nil {
 		return false, err
 	}
-	if _, err := getNodeAgentResourceReqs(dpa); err != nil {
+	if _, err := getNodeAgentResourceReqs(r.dpa); err != nil {
 		return false, err
 	}
 
 	// validate non-admin enable and tech-preview-ack
-	if r.checkNonAdminEnabled() {
-		if !(dpa.Spec.UnsupportedOverrides[oadpv1alpha1.TechPreviewAck] == TrueVal) {
+	if r.dpa.Spec.NonAdmin != nil && r.dpa.Spec.NonAdmin.Enable != nil && *r.dpa.Spec.NonAdmin.Enable {
+		if !(r.dpa.Spec.UnsupportedOverrides[oadpv1alpha1.TechPreviewAck] == TrueVal) {
 			return false, errors.New("in order to enable/disable the non-admin feature please set dpa.spec.unsupportedOverrides[tech-preview-ack]: 'true'")
 		}
+
+		dpaList := &oadpv1alpha1.DataProtectionApplicationList{}
+		err = r.ClusterWideClient.List(r.Context, dpaList)
+		if err != nil {
+			return false, err
+		}
+		for _, dpa := range dpaList.Items {
+			if dpa.Namespace != r.NamespacedName.Namespace && (&DPAReconciler{dpa: &dpa}).checkNonAdminEnabled() {
+				nonAdminDeployment := &appsv1.Deployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      nonAdminObjectName,
+						Namespace: dpa.Namespace,
+					},
+				}
+				if err := r.ClusterWideClient.Get(
```

Comment:
Shouldn't we be using the older ns-scoped client here ?

Reply:
no

the check here is to check if other namespace, different than the namespace this OADP is installed, has NAC installed already. We check that by getting DPAs in other namespaces, and on those namespaces, if nonAdmin is enabled, check if nonAdmin Deployment exist. So both calls need cluster wide privileges